src/losses/Losses.jl

```diff
@@ -1,54 +1,404 @@
+            normalization = normalization,
+            )
+        for (ai, bi) in zip(a, b)
+        ]
+    # return [sum(( (ai .- bi)[is_in_glacier(bi, lossType.distance)] ).^2)/normalization for (ai,bi) in zip(a,b)]
 end
-function backward_loss(lossType::L2Sum, a::Vector{Matrix{F}}, b::Vector{Matrix{F}}; normalization::F=1.) where {F <: AbstractFloat}
+function backward_loss(
+    lossType::L2Sum,
+    a::Vector{Matrix{F}},
+    b::Vector{Matrix{F}};
+    normalization::F=1.,
+) where {F <: AbstractFloat}
     @assert length(a) == length(b) "Size of a and b don't match: length(a)=$(length(a)) but length(b)=$(length(b))"
-    return [backward_loss(lossType, ai, bi; normalization=normalization) for (ai,bi) in zip(a,b)]
+    return [
+        backward_loss(
+            lossType,
+            ai,
+            bi;
+            normalization=normalization,
+            )
+        for (ai, bi) in zip(a, b)
+        ]
 end
 
-function loss(lossType::L2SumWithinGlacier, a::Vector{Matrix{F}}, b::Vector{Matrix{F}}; normalization::F=1.) where {F <: AbstractFloat}
-    @assert length(a) == length(b) "Size of a and b don't match: length(a)=$(length(a)) but length(b)=$(length(b))"
-    return [sum(( (ai .- bi)[is_in_glacier(bi, lossType.distance)] ).^2)/normalization for (ai,bi) in zip(a,b)]
+### Definition of Loss functions
+
+"""
+    function loss(
+        lossType::LogSum,
+        a::Matrix{F},
+        b::Matrix{F},
+        mask::BitMatrix;
+        normalization::F=1.,
+    ) where {F <: AbstractFloat}
+
+Compute logaritmic loss function for ice velocity fields following Morlighem, M. et al.,
+"Spatial patterns of basal drag inferred using control methods from a full-Stokes and
+simpler models for Pine Island Glacier, West Antarctica". Geophys. Res. Lett. 37, (2010).
+Given a minimum velocity ϵ the absolute velocity given by a and b, it computes the sum of
+
+    log^2( (a + ϵ) / (b + ϵ) )
+
+It has been shown that this loss function enables robust estimation of drag coefficient.
+"""
+function loss(
+    lossType::LogSum,
+    a::Matrix{F},
+    b::Matrix{F},
+    mask::BitMatrix;
+    normalization::F=1.,
+) where {F <: AbstractFloat}
+    @assert (minimum(a) >= 0.0) & (minimum(b) >= 0.0)
+    return sum((log.((a .+ lossType.ϵ) ./ (b .+ lossType.ϵ)).^2)[mask]) ./ normalization
 end
-function backward_loss(lossType::L2SumWithinGlacier, a::Vector{Matrix{F}}, b::Vector{Matrix{F}}; normalization::F=1.) where {F <: AbstractFloat}
-    @assert length(a) == length(b) "Size of a and b don't match: length(a)=$(length(a)) but length(b)=$(length(b))"
-    return [backward_loss(lossType, ai, bi; normalization=normalization) for (ai,bi) in zip(a,b)]
+function backward_loss(
+    lossType::LogSum,
+    a::Matrix{F},
+    b::Matrix{F},
+    mask::BitMatrix;
+    normalization::F,
+) where {F <: AbstractFloat}
+    d = zero(a)
+    d[mask] = log.((a[mask] .+ lossType.ϵ) ./ (b[mask] .+ lossType.ϵ)) ./ (b[mask] .+ lossType.ϵ)
```

Comment:
Have you tested this backward? Assuming that the function is being differentiated wrt `a`, when I write down the equations I have a different formula.
```suggestion
    d[mask] = log.((a[mask] .+ lossType.ϵ) ./ (b[mask] .+ lossType.ϵ)) ./ (a[mask] .+ lossType.ϵ)
```

Reply:
By the way we should test this new loss function by adding a test in the `"Manual backward of the loss terms vs Enzyme"` testset

Reply:
This is embarrassing @albangossard ... all this automatic differentiation and I made a simple manual differentiation error. I am a shame, you should stop collaborating with me... I fix this now and tests are passing XD